### DISCUSSION TITLE
New posts detection a bit janky?

### DISCUSSION THREAD

**Discussion by Geobert** [cheeaun/phanpy#580] [Q&A] [answered]
Hi,

When running side by side Phampy and another front (the original or elk for instance), it seems that phampy is a bit slow to detect that new posts are available.

Is that a known issue?

**Comment by cheeaun** [comment, marked as answer]
@Geobert it should be a bit slower than Elk and original web site as it's not using streaming API.

**Reply by Geobert** [reply]
I see, out of curiosity, why? :)

**Reply by cheeaun** [reply]
It was buggy previously:

- Some servers have flaky streaming API — I'm not sure if it's fixed now, there are too many AP servers out there to test.
- Sometimes new posts are available earlier than the (server-cached) REST API — I think it's fixed now, so will need to revisit later.
- Streaming, when there's a crazy huge amount of posts, can get pretty messy, affecting performance, memory usage and even usability.
- Streaming (via masto.js library) were not multiplexed and not sharing one WebSocket connection — Now it's fixed in masto.js v6, so will need to revisit.
- Streaming for multiple timelines in multi-column UI will probably introduce even more performance issues, so will need to experiment first.

**Reply by Geobert** [reply]
That makes perfect sense, thanks!
